### PR TITLE
docs: sync onboarding docs with shipped v0.9.0 CLI and Flow schema

### LINKED ISSUE #189: README roadmap shows v0.4.0 as the current shipped milestone, but the project is at v0.7.0
**Type:** Documentation
**Priority:** P1
**Newcomer impact:** A newcomer looking at the roadmap table is told "current" is v0.4.0 ("Add Async, Persistence & Visualization") and that v0.5 (Schema Governance), v0.6 (Integrations), v0.7 (CLI & Benchmarks) are still "planned". In reality, all three have shipped per `CHANGELOG.md`. This causes new users to underestimate what's available (e.g., they won't think to look for `chainweaver attest` or `chainweaver suggest`).

## Evidence

- File/path/doc section: `README.md` lines 506–515 (roadmap table).
- Actual state: `pyproject.toml:7` → 0.7.0; `CHANGELOG.md` records 0.5.0, 0.6.0, 0.7.0 entries; commit `baccf87 chore: release v0.7.0` already merged.
- Snippet from README:
  > **v0.4.0** … `shipped (current)`
  > **v0.5.0** … `planned`
  > **v0.6.0** … `planned`
  > **v0.7.0** … `planned`

## Steps to reproduce

1. Read `README.md` § Roadmap.
2. Compare with `CHANGELOG.md` and `pyproject.toml`.

## Proposed fix

Update the roadmap table so v0.5.0, v0.6.0, and v0.7.0 are marked `shipped`, v0.7.0 (or whatever is next on the milestones page) is `current`, and the next planned milestone is marked accordingly. Cross-link to `CHANGELOG.md` for the per-release breakdown.

## Implementation notes for AI

- Likely files to inspect/change: `README.md` lines 506–515.
- Relevant tests to add or update: optionally add a release-checklist item in `CONTRIBUTING.md` to bump this table when a new version is published.
- Edge cases: if the team prefers not to keep statuses inline, link out to GitHub Milestones instead and stop hand-maintaining the column.

## Acceptance criteria

- [ ] Roadmap table reflects actual ship status as of v0.7.0.
- [ ] CHANGELOG, pyproject, and roadmap agree on the current version.
- [ ] No "planned" rows for milestones that have already shipped.

## Duplicate check

Searched open issues for: "roadmap", "v0.4.0 current", "milestone status", "shipped", "outdated roadmap".
Result: no open duplicate found.

### PR DESCRIPTION
**Pull request by dgenio**
## Summary

The onboarding docs drifted behind the shipped **v0.9.0** surface: the roadmap still listed v0.4.0 as "current", the canonical `Flow` field table omitted required fields, the architecture map described the CLI as a single `inspect` command, and the CLI reference never documented the required `type:` discriminator, the `[yaml]` extra, or the `attest` / `suggest` / `dump-schema` commands. This PR resolves that cluster of doc-drift issues in one pass and adds anti-drift guards so they don't recur.

Docs-only change plus one runnable example flow file and one new test — **no `chainweaver/` source code touched**, so runtime behavior is unchanged.

## Changes

- **`README.md`**
  - Roadmap table: v0.4.0–v0.9.0 marked `shipped`, current is v0.9.0, added v0.8.0/v0.9.0 rows (#189).
  - Architecture `Flow` snippet gains `version=` and points to the full field table (#192).
  - CLI section: `run` example now uses the shipped `examples/double_add_format.flow.yaml`; documents the `type:` discriminator and the `chainweaver[yaml]` extra (#195, #196).
- **`AGENTS.md`** — §5 `Flow` field table rewritten to match `Flow.model_fields` exactly: adds `version`, `status`, `tool_schema_hashes`, renames `input_schema`/`output_schema` to the `*_schema_ref` string fields, and notes the resolved-property accessors (#192).
- **`docs/agent-context/architecture.md`** — `cli.py` row now describes the full 11-command surface and the disk-vs-registry split (#193).
- **`docs/cli.md`** — new **"Flow file format"** section (the `type:` discriminator + `chainweaver[yaml]` extra, #195/#196); new `### attest`, `### suggest`, `### dump-schema` reference sections matching the existing template (#197); programmatic-registration footer updated.
- **`examples/double_add_format.flow.yaml`** *(new)* — minimal, valid, runnable `type: Flow` file referenced by the README/CLI quick start (#195).
- **`tests/test_cli_docs.py`** *(new)* — anti-drift guards: every registered CLI command has a `docs/cli.md` section; the shipped example validates and runs end-to-end.

**Mode B note:** the issues only named `attest`/`suggest` for #197, but I also documented `dump-schema` (the third undocumented command) so the new command-coverage test passes for all 11 commands and the CLI reference can't silently drift again — directly serving the #193/#197 intent.

## Testing

- [x] Linting passes (`ruff check chainweaver/ tests/ examples/`) → `All checks passed!`
- [x] Formatting check passes (`ruff format --check chainweaver/ tests/ examples/`) → `149 files already formatted`
- [x] Type checking passes (`python -m mypy chainweaver/`) — new `tests/test_cli_docs.py` is `Success: no issues found`. The 8 `import-not-found` errors in my sandbox are pre-existing and confined to optional-extra modules (`mcp`, `opentelemetry`, `llama_index`, `langchain`) not installed here; none are in files this PR touches, and CI installs `[dev]`.
- [x] All existing tests pass — ran 237 tests across the affected areas (`tests/test_cli*.py`, `test_versioning.py`, `test_serialization.py`, `test_flow_execution.py`, `test_flow_status.py`, `test_flow_capability.py`) → all green; the 4 new `test_cli_docs.py` tests pass.
- [x] New tests added for new functionality
- [x] `mkdocs build --strict` → exit `0` (verified the new `docs/cli.md` links use the full-GitHub-URL convention for `AGENTS.md`, since it sits outside `docs_dir`).

Manual end-to-end check of the shipped example:

```
$ chainweaver validate examples/double_add_format.flow.yaml
OK  examples/double_add_format.flow.yaml: double_add_format v0.1.0 [Flow]   # exit 0
$ chainweaver run examples/double_add_format.flow.yaml --tools examples.simple_linear_flow --input '{"number": 5}'
final_output: {"number": 5, "value": 20, "result": "Final value: 20"}      # exit 0
```

## Related Issues

Closes #189, #192, #193, #195, #196, #197.

(Sibling doc-drift issues #190 and #194 are intentionally out of scope — they already have open PRs #199 and #198 respectively.)

## Checklist
- [x] Code follows project conventions (see `AGENTS.md` and `docs/agent-context/`)
- [x] Public API changes are documented — N/A (no public API change; docs + example + test only)
- [x] No secrets or credentials included

## Risks / caveats

- Docs/example/test only — no executor or public-API change, so the three hard executor invariants are untouched.
- The documented `chainweaver run examples/...` command requires running from the repo root so the `examples` namespace package is importable (same convention as `python examples/<name>.py`); the doc comment states this and the new test verifies it.

---
_Generated by [Claude Code](https://claude.ai/code/session_01LSc3SzeNeJgvdJ7hgP4Bff)_

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -749,7 +756,9 @@ chainweaver doctor flows/ --check-drift --tools my_pkg.tools
 `run` is the fastest path from a fresh install to seeing a flow execute:
 point it at a `.flow.yaml`/`.flow.json` file, pass `--tools <module>` (the
 import path of a Python module that exposes `Tool` instances at top
-level), and supply the initial input as JSON. Every subcommand also
+level), and supply the initial input as JSON. Hand-authored flow files must
+declare a `type: Flow` (or `type: DAGFlow`) discriminator at the top — see
+the [flow file format](docs/cli.md#flow-file-format) reference. Every subcommand also
 supports `--format json` for machine consumption, and shares the same
 exit-code contract (`0` success, `1` business-logic error, `2`
 file-not-found / argument error).
```
[the remark this answers is not in the document]
Comment (dgenio) [reply README.md]: Good catch — fixed in 23cb5f2.

The sentence now enumerates the actual taxonomy: nine commands accept `--format table|json` (`inspect`, `validate`, `check`, `run`, `profile`, `diff`, `attest`, `suggest`, `doctor`), with the two exceptions called out explicitly — `viz` uses `--format ascii|dot`, and `dump-schema` writes a raw JSON Schema and has no `--format` flag. The exit-code-contract sentence stays since that part really is shared across all subcommands.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: AGENTS.md [hunk 2]
```diff
@@ -172,18 +172,30 @@ For the full prohibited-actions list and anti-patterns, see
 
 ### `Flow` (Pydantic model)
 
+The rows below are the actual Pydantic fields (`Flow.model_fields`), in
+declaration order. `DAGFlow` carries the same fields except `version` is
+**required** (no default).
```
[the remark this answers is not in the document]
Comment (dgenio) [reply AGENTS.md]: Good catch — fixed in 23cb5f2.

The qualifying paragraph above the field table now reads: `DAGFlow` carries the same field **names**, with two differences — `version` is **required** (no default), and `steps` is typed `list[DAGFlowStep]` (which extends `FlowStep` with conditional `branches`), with a forward-link to the `DAGFlowStep` subsection already present right below. Verified against `chainweaver/flow.py:453,812`.

---
_Generated by [Claude Code](https://claude.ai/code)_
